lib/engine/step/g_1846/buy_sell_par_shares.rb

```diff
@@ -7,15 +7,31 @@ module Step
     module G1846
       class BuySellParShares < BuySellParShares
         def can_buy?(entity, bundle)
-          if @game.bundle_is_presidents_share_alone_in_pool?(bundle)
-            bundle = bundle.to_bundle
-            return false unless entity.num_shares_of(bundle.corporation) == 1
+          return unless bundle
 
-            percent = 10
-            bundle = ShareBundle.new(bundle.shares, percent)
+          bundle = bundle.to_bundle
+          corporation = bundle.corporation
+
+          if corporation.receivership? && bundle.presidents_share
+            return false if entity.num_shares_of(corporation).zero?
+
+            bundle = ShareBundle.new(bundle.shares, 10)
+          end
+
+          super(entity, bundle)
+        end
+
+        def process_buy_shares(action)
+          bundle = action.bundle
```

Comment:
return unless (bundle = action.bundle)

rather than all the safe operators

Reply:
That's what I initially had; however, it prevents the parent from being called, which adds the game error to be displayed to the player. Two Options come to mind.

1) Handle the error message here. This duplicates functionality handled later.
@game.game_error("Cannot buy a share of #{shares&.corporation&.name}") unless (bundle = action.bundle)

2) Call super early and let the parent handle the error.
return super unless (bundle = action.bundle)

Reply:
ah you're right. let's do number 2

Reply:
Option 2 implemented.